src/pages/ProfileDisplay/Profile.jsx

```diff
@@ -1,47 +1,133 @@
 import { useState, useEffect } from "react";
-import "./Profile.css";
-import { getLocations } from "../../services/locations";
-import LocationForm from "../../components/LocationForm/LocationForm";
-import LocationTile from "../../components/LocationTile/LocationTile";
-import LogoutButton from "../../components/NavButtons/LogoutButton";
+import { getLocations } from "../../services/locations";  // Fetch locations from API
+import { getItems } from "../../services/items";  // Fetch items by locationId from API
+import { Link } from "react-router-dom";  // Import Link from react-router-dom
+import LocationTile from "../../components/LocationTile/LocationTile";  // Render each location
+import ItemTile from "../../components/ItemTile/ItemTile";  // Render each item
+import LocationForm from "../../components/LocationForm/LocationForm";  // Form to add locations
+import LogoutButton from "../../components/NavButtons/LogoutButton";  // Logout functionality
+import SearchBar from "../../components/SearchBar/SearchBar";  // Search bar component
+import './Profile.css';  // Import CSS for styling
 
 const Profile = ({ setIsSignedIn }) => {
-  const [locations, setLocations] = useState([]);
-  const [showForm, setShowForm] = useState(false); // State to show/hide the form
-
+  const [locations, setLocations] = useState([]);  // State for locations
+  const [allItems, setAllItems] = useState({});  // Stores items for each location
+  const [filteredLocations, setFilteredLocations] = useState([]);  // State for filtered locations
+  const [filteredItems, setFilteredItems] = useState([]);  // Only show items when searched
+  const [searchQuery, setSearchQuery] = useState("");  // State for search query
+  const [showForm, setShowForm] = useState(false);  // Toggle form visibility
+  const [selectedLocationId, setSelectedLocationId] = useState(null);  // For editing location
+  
   useEffect(() => {
-    const loadLocations = async () => {
-      const userLocations = await getLocations();
-      setLocations(userLocations);
+    const loadData = async () => {
+      try {
+        const userLocations = await getLocations();
+        setLocations(userLocations);
+        setFilteredLocations(userLocations);
+
+        // Fetch items for each location and assign location_id to each item
+        const itemsForLocations = {};
+        for (let location of userLocations) {
+          const items = await getItems(location.id);  // Fetch items for each location
+          
+          // Ensure each item has the location_id assigned
+          itemsForLocations[location.id] = items.map(item => ({
+            ...item,
+            location_id: location.id  // Assign location_id to each item
+          }));
+        }
+        setAllItems(itemsForLocations);  // Store items for each location
```

Comment:
Maybe search and filter items should live on the location display page where we list out the items?